### DISCUSSION TITLE
Packaging to EXE with Auto-Py-to_Exe.... Error on Open

### DISCUSSION THREAD

**Discussion by oh92891** [TomSchimansky/CustomTkinter#896] [General]
Hey there I followed the instructions on the packaging page, and the file packages correctly, however when I try to run the EXE the following error is generated: 

Traceback (most recent call last):
  File "main.py", line 1, in <module>
  File "C:\Users\XXXXX\PycharmProjects\XXXXX\output\main\customtkinter\__init__.py", line 9, in <module>
    import tkinter.filedialog as filedialog
ModuleNotFoundError: No module named 'tkinter.filedialog'

Please help, thank you!

**Comment by Akascape** [comment]
@oh92891 Add the tkinter folder in your app manually using additional folder option in auto-py-to-exe.
The path for tkinter module is: `Python\Lib\tkinter`

**Reply by oh92891** [reply]
Thanks! Just did that and now receiving this error when attempting to run the .EXE:

Traceback (most recent call last):
  File "main.py", line 5, in <module>
  File "C:\Users\XXXXX\PycharmProjects\XXXXX\output\main\customtkinter\__init__.py", line 7, in <module>
    import tkinter.filedialog as filedialog
  File "C:\Users\XXXXX\PycharmProjects\XXXXX\output\main\tkinter\filedialog.py", line 3, in <module>
    from future.utils import PY3
ModuleNotFoundError: No module named 'future'

I eventually need to package the GUI in to an install-able file but I am not sure it will be possible?

**Reply by Akascape** [reply]
@oh92891 I guess your pyinstaller is unable to recognize the modules because of different python environments.

**Reply by Akascape** [reply]
@oh92891 The error seems to be, you are using modules which are installed in venv but you are trying to convert to .exe using system interpreter. Or maybe the opposite is happening.

**Reply by oh92891** [reply]
Is there any way to fix that and still create an installable executable?

I apologize for my basic questions, I am new to the whole Python language and just trying to learn.

**Reply by Akascape** [reply]
@oh92891 Install pyinstaller (Auto-py-to-exe) on your native python instead of pycharm. (pip install and use that)

**Reply by oh92891** [reply]
Would that be doing the process through CMD? I just tried that and got the same errors unfortunately.

**Reply by Akascape** [reply]
@oh92891 Try `python -m pip install auto-py-to-exe`. 
Then open this file in the path: `Python\Scripts\auto-py-to-exe.exe`.

**Reply by oh92891** [reply]
Same error when opening the new EXE: 
Traceback (most recent call last):
  File "main.py", line 5, in <module>
  File "C:\Users\XXXXX\AppData\Local\Programs\Python\Python310\Scripts\output\main\customtkinter\__init__.py", line 7, in <module>
    import tkinter.filedialog as filedialog
  File "C:\Users\XXXXX\AppData\Local\Programs\Python\Python310\Scripts\output\main\tkinter\filedialog.py", line 3, in <module>
    from future.utils import PY3
ModuleNotFoundError: No module named 'future'

**Reply by Akascape** [reply]
@oh92891 Ok, I guess this will be the fix: https://stackoverflow.com/a/36618847
Uncheck those options in pycharm.
